### PR TITLE
Ensure the shared framework build uses the generated runtime.json

### LINKED ISSUE #53550: Ensure shared framework consumes source-built RID graph
Related to https://github.com/dotnet/runtime/pull/50818

I believe the shared framework build process is still consuming the statically defined, checked-in, runtime.json.

We need to make sure it uses the generated one.

I suggest we do this by making the shared framework always consume it from an output location (along with the assemblies, etc).  Then make the `Microsoft.NETCore.Platforms` project responsible for putting it there, whether it builds it or not.

Also we should consider scenarios like https://github.com/dotnet/runtime/pull/52907 where folks want to add RIDs without necessarily doing a source build.

Perhaps there is a way to use the RID generation process in more conditions while still ensuring the checked-in copy is up-to-date?

### PR DESCRIPTION
**Pull request by ericstj**
Fixes https://github.com/dotnet/runtime/issues/53550

### CODE REVIEW

File: eng/liveBuilds.targets [hunk 1]
```diff
@@ -221,6 +221,6 @@
             ResolveLibrariesRuntimeFilesFromLocalBuild" />
 
   <PropertyGroup>
-    <BundledRuntimeIdentifierGraphFile>$(RuntimeIdGraphDefinitionFile)</BundledRuntimeIdentifierGraphFile>
+    <BundledRuntimeIdentifierGraphFile>$(LiveRuntimeIdGraphDefinitionFile)</BundledRuntimeIdentifierGraphFile>
```
Comment (ViktorHofer) [inline eng/liveBuilds.targets:224]: By switching to the live generated runtime.json file, you are depending on the file already being generated which isn't the case until the Microsoft.NETCore.Platforms project is built. We should make sure that any code path that depends on it has a dependency to the Platforms project.

**Comment by ericstj** [comment]
Closing this as I don't have cycles to drive it at the moment.  @tmds feel free to fix this as it best fits the work you're doing.

**Comment by tmds** [comment]
@ericstj the minimal thing we need for source-build is: have the non-portable rid show up in `$DOTNET_ROOT/shared/Microsoft.NETCore.App/7.0.0/Microsoft.NETCore.App.deps.json`.
Do you know where this file gets generated? Can we apply a patch so that picks up the `runtime.json` file from `Microsoft.NETCore.Platforms.csproj`?

**Comment by ericstj** [comment]
There are two places that generate a shared framework deps file.  One is `pretest.proj` which is used by library testing.  
https://github.com/dotnet/runtime/blob/1f1231c36a9b247aaf0111496d3c0a3b3611ef15/src/libraries/pretest.proj#L97

The other is the shared framework SDK which uses the `BundledRuntimeIdentifierGraphFile` property https://github.com/dotnet/arcade/blob/9bbfa8dbc89c406eb8e2bc62600a007882ccd0ab/src/Microsoft.DotNet.SharedFramework.Sdk/targets/sharedfx.targets#L197

**Pull request closed without merging**